app/controllers/Organizations.scala

```diff
@@ -90,26 +87,24 @@ class Organizations @Inject()(forms: OreForms,
     */
   def setInviteStatus(id: Int, status: String) = Authenticated.async { implicit request =>
     val user = request.user
-    user.organizationRoles.get(id).flatMap {
-      case None => Future.successful(notFound)
-      case Some(role) =>
-        role.organization.map { orga =>
-          status match {
-            case STATUS_DECLINE =>
-              orga.memberships.removeRole(role)
-              Ok
-            case STATUS_ACCEPT =>
-              role.setAccepted(true)
-              Ok
-            case STATUS_UNACCEPT =>
-              role.setAccepted(false)
-              Ok
-            case _ =>
-              BadRequest
-          }
+    user.organizationRoles.get(id).semiFlatMap { role =>
+      //TODO: Why access the organization when it's only used for one of the statuses?
```

Comment:
That's a bit strange. I think we add the user to the organization before he has accepted this invite. In order to this we have to remove him when he declined and that's why we have to access to the organization here.